### DISCUSSION TITLE
`immich upload` cli command finds no files in directories with single quotes in their names

### DISCUSSION THREAD

**Discussion by frebib** [immich-app/immich#12076] [Q&A]
### The bug

```
/import # immich upload -c32 -r -a 2020-02-02\ Joe\'s\ Birthday/
Crawling for assets...
No files found, exiting

/import # ls -lFAh 2020-02-02\ Joe\'s\ Birthday/
total 9G
-rw-rw-rw-    1 node     node       35.4M Feb  1  2020 IMG_4075.CR2
-rw-r--r--    1 node     node         985 Nov 28  2021 IMG_4075.CR2.xmp
-rw-rw-rw-    1 node     node        8.4M Feb  1  2020 IMG_4075.JPG
-rw-r--r--    1 node     node         985 Nov 28  2021 IMG_4075.JPG.xmp
...
```
I can manually import them the hard way though
```
/import/2020-02-02 Joe's Birthday # immich upload -c32 -r -A "Joe's Birthday 2020" *.JPG *.CR2
Crawling for assets...
Checking files | ████████████████████████████████████████ | 100% | ETA: 0s | 396/396 assets
Found 396 new files and 0 duplicates
Uploading assets | ████████████████████████████████████████ | 100% | ETA: 0s | 9.4 GB/9.4 GB
Successfully uploaded 396 new assets (9.4 GB)
Creating albums | ████████████████████████████████████████ | 100% | ETA: 0s | 1/1 albums
Successfully created 1 new album
Successfully updated 396 assets
Adding assets to albums | ████████████████████████████████████████ | 100% | ETA: 0s | 396/396 assets
```

I had this issue trying to import several albums- all that failed had single quotes in

### The OS that Immich Server is running on

Debian, in Docker

### Version of Immich Server

1.112.1

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
-
```

### Your .env content

```Shell
-
```

### Reproduction steps

See `The bug` at the top

**Comment by alextran1502** [comment]
I think this is how Linux parses space and special characters

**Reply by frebib** [reply]
The input was correctly escaped- I don't think this is *nix/*sh's fault

**Reply by frebib** [reply]
Can you point me at the tests for this code? https://github.com/immich-app/immich/blob/main/cli/src/utils.ts#L115

**Reply by desmondgc** [reply]
@frebib You can look at [utils.spec.ts](https://github.com/immich-app/immich/blob/main/cli/src/utils.spec.ts) and [upload.e2e-spec.ts](https://github.com/immich-app/immich/blob/main/e2e/src/cli/specs/upload.e2e-spec.ts).

I have the same issue on Windows running cli v2.2.37 (server: Immich v1.123.0).

**Comment by mmomjian** [comment]
Fixed in #14856

**Reply by desmondgc** [reply]
Not quite. See https://github.com/immich-app/immich/pull/15283.

**Reply by mmomjian** [reply]
Interesting. Thanks!
